flow/project.py

```diff
@@ -914,10 +914,11 @@ def _fetch_scheduler_status(self, jobs=None, file=None, ignore_errors=False):
             scheduler_info = {sjob.name(): sjob.status() for sjob in self.scheduler_jobs(scheduler)}
             status = dict()
             print(self._tr("Query scheduler..."), file=file)
-            for op in tqdm(self._job_operations(jobs=jobs, only_eligible=False),
-                           desc="Fetching operation status",
-                           total=len(jobs), file=file):
-                status[op.get_id()] = int(scheduler_info.get(op.get_id(), JobStatus.unknown))
+            for job in tqdm(jobs,
+                            desc="Fetching operation status",
+                            total=len(jobs), file=file):
+                for op in self._job_operations(jobs=[job], only_eligible=False):
```

Comment:
Can you refactor the `_job_operations()` function to accept a single `job` argument instead of a sequence of jobs? I believe that the function accepts a sequence right now to prepare us for aggregation-operations, but the loop-structure as implemented right now simply represents an anti-pattern.

No need to accommodate for backwards-compatibility, since this is a purely internal function.

Reply:
Agreed, I was considering that as a possible solution and i think it's the right call.